spyder/widgets/findinfiles.py

```diff
@@ -203,21 +206,181 @@ def get_results(self):
-        self.path = path
-        QListWidgetItem.__init__(self, self.__repr__(), parent)
+class SearchInComboBox(QComboBox):
+    """
+    Non editable combo box handling the path locations of the FindOptions
+    widget.
+    """
+    def __init__(self, external_path_history=[], parent=None):
+        super(SearchInComboBox, self).__init__(parent)
+        self.setSizePolicy(QSizePolicy.Expanding, QSizePolicy.Fixed)
+        self.setToolTip(_('Search directory'))
+        self.setEditable(False)
 
-    def __repr__(self):
-        if len(self.path) > MAX_PATH_LENGTH:
-            return truncate_path(self.path)
-        return self.path
+        self.path = ''
+        self.project_path = None
+        self.file_path = None
+        self.external_path = None
 
-    def __str__(self):
-        return self.__repr__()
+        self.addItem(_("Current working directory"))
+        ttip = ("Search in all files and directories present on the current"
+                " Spyder path")
+        self.setItemData(0, ttip, Qt.ToolTipRole)
 
-    def __unicode__(self):
-        return self.__repr__()
+        self.addItem(_("Project"))
+        ttip = _("Search in all files and directories present on the"
+                 " current project path (if opened)")
+        self.setItemData(1, ttip, Qt.ToolTipRole)
+        self.model().item(1, 0).setEnabled(False)
+
+        self.addItem(_("File").replace('&', ''))
+        ttip = _("Search in current opened file")
+        self.setItemData(2, ttip, Qt.ToolTipRole)
+
+        self.insertSeparator(3)
+
+        self.addItem(_("Select other directory"))
+        ttip = _("Search in other folder present on the file system")
+        self.setItemData(4, ttip, Qt.ToolTipRole)
+
+        self.addItem(_("Clear this list"))
+        ttip = _("Clear the list of other directories")
+        self.setItemData(5, ttip, Qt.ToolTipRole)
+
+        self.insertSeparator(6)
+
+        for path in external_path_history:
+            self.add_external_path(path)
+
+        self.currentIndexChanged.connect(self.path_selection_changed)
+        self.view().installEventFilter(self)
+
+    def add_external_path(self, path):
+        """
+        Adds an external path to the combobox if it exists on the file system.
+        If the path is already listed in the combobox, it is removed from its
+        current position and added back at the end. If the maximum number of
+        paths is reached, the oldest external path is removed from the list.
+        """
+        if not os.path.exists(path):
+            return
+        self.removeItem(self.findText(path))
+        self.addItem(path)
+        self.setItemData(self.count()-1, path, Qt.ToolTipRole)
+        while self.count() > MAX_PATH_HISTORY+EXTERNAL_PATHS:
+            self.removeItem(EXTERNAL_PATHS)
+
+    def get_external_paths(self):
+        """Returns a list of the external paths listed in the combobox."""
+        return [to_text_string(self.itemText(i))

```

Comment:
We probably need to use here `to_unicode_from_fs` (from `utils/encoding.py`) to deal with non-ascii dirs on Windows that uses encodings like cp-1251 (Russian).

Reply:
Sorry, this is not the proper place to do it. I'll point to the right place below.

Reply:
Should I replace `to_text_string(self.itemText(i)` by `self.itemText(i)`?

I do not understand what `to_text_string` is doing here since `self.itemText(i)` is already returning a unicode str by default as stated, for instance, in the [PySide documentation](https://srinikom.github.io/pyside-docs/PySide/QtGui/QComboBox.html#PySide.QtGui.PySide.QtGui.QComboBox.itemText)). Or is it returning a `QString` in some version of PyQt on python 2?

All this encoding stuff is mumbo jumbo to me, no matter how many times I've read on this subject XD.

Reply:
Ok, I've seen that `to_text_string` was always used with `itemText` everywhere else. So maybe it is safer to keep it that way.

Reply:
OK, I've read that Python 2, PyQt API #1 is returning a `QString`, so I'll keep the `to_text_string`.

Reply:
Ok, good idea.